src/components/CvssCalculator/CvssCalculatorOverlayed.vue

```diff
@@ -0,0 +1,164 @@
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue';
+
+import CvssVectorInput from '@/components/CvssCalculator/CvssVectorInput.vue';
+import CvssFactorBtns from '@/components/CvssCalculator/CvssFactorBtns.vue';
+
+import {
+  getFactors,
+  calculateScore,
+  formatFactors,
+  validateCvssVector,
+} from '@/composables/useCvssCalculator';
+
+const cvssVector = defineModel<null | string | undefined>('cvssVector');
+const cvssScore = defineModel<null | number | undefined>('cvssScore');
+
+const emit = defineEmits<{
+  updateAffectCvss: [vector: string, score: null | number];
+}>();
+
+const error = computed(() => validateCvssVector(cvssVector.value));
+const cvssFactors = ref<Record<string, string>>({});
+const isFocused = ref(false);
+
+const cvssDiv = ref();
+const cvssVectorInput = ref();
+
+function updateFactors(newCvssVector: null | string | undefined) {
+  if (cvssVector.value !== newCvssVector) {
+    emit('updateAffectCvss', newCvssVector || '', calculateScore(cvssFactors.value) || null);
+  }
+  cvssFactors.value = getFactors(newCvssVector ?? '');
+}
+
+updateFactors(cvssVector.value);
+
+watch(() => cvssVector.value, () => {
+  updateFactors(cvssVector.value);
+  emit('updateAffectCvss', cvssVector.value || '', calculateScore(cvssFactors.value) || null);
+});
+
+function onInputFocus(event: FocusEvent) {
+  isFocused.value = true;
+  if (event.target !== cvssVectorInput.value.input) {
+    cvssVectorInput.value.input.focus();
+  }
+}
+
+function onInputBlur(event: FocusEvent) {
+  if (event.relatedTarget !== cvssDiv.value) {
+    isFocused.value = false;
+  }
+}
+
+function reset() {
+  cvssScore.value = null;
+  emit('updateAffectCvss', '', null);
```

Comment:
Should this be `emit('affect:cvss:update', '', null)`?

Reply:
As commented off thread, it needs to be discussed what convention we get into and apply it to all emits